### DISCUSSION TITLE
Trying to understand serial communication.

### DISCUSSION THREAD

**Discussion by jwinfieldnz** [luc-github/ESP3D#797] [General]
Great project - thanks for making it. 

I am using the 8888 port to print directly from octoprint to my E3V2. When using an ESP8266 the blobs were quite bad. I am trying an ESP32 now and it doesn't seem as bad. I kinda know this is network latency etc but I am just wondering why it is not possible to have some sort of FIFO buffer that just fills up when the job starts. Or is that actually happening and the ram size just is big enough on the ESP32 to overcome what the printer is consuming?

Sorry, there is probably an obvious answer to this. 

Keep up the good work.

**Comment by luc-github** [comment]
the buffer you are  refering is in printer FW , ESP3D just do the bridge and do not buffer anything.
 ESP32 is better because it has 2 cores and higher frequency
